src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -165,7 +165,7 @@ private static unsafe int Send(SafeCloseSocket socket, SocketFlags flags, IList<
                 for (int i = 0; i < maxBuffers; i++, startOffset = 0)
                 {
                     ArraySegment<byte> buffer = buffers[startIndex + i];
-                    Debug.Assert(buffer.Offset + startOffset < buffer.Array.Length, $"Unexpected values: Offset={buffer.Offset}, startOffset={startOffset}, Length={buffer.Array.Length}");
+                    RangeValidationHelpers.ValidateSegment(buffer);
```

Comment:
How so?  No matter what we do, we're being given an interface-based collection and we have to extract its contents, so there could certainly be a tear while reading out of the collection, that's unavoidable.  But that's also why it's stored into a local first, and then we validate and use the local, so if we do get torn data, we validate that torn data and then use it.  I don't see any way to do better.  What do you suggest?